src/functTest/groovy/com/bmuschko/gradle/docker/AbstractFunctionalTest.groovy

```diff
@@ -33,26 +33,21 @@ abstract class AbstractFunctionalTest extends Specification {
     def setup() {
         projectDir = temporaryFolder.root
         buildFile = temporaryFolder.newFile('build.gradle')
-
-        def pluginClasspathResource = getClass().classLoader.findResource("plugin-classpath.txt")
-        if (pluginClasspathResource == null) {
-            throw new IllegalStateException("Did not find plugin classpath resource, run 'functionalTestClasses' build task.")
-        }
-
-        def pluginClasspath = pluginClasspathResource.readLines()
-                .collect { it.replace('\\', '\\\\') } // escape backslashes in Windows paths
-                .collect { "'$it'" }
-                .join(", ")
-
-        // Add the logic under test to the test build
+                
+        def pluginClasspath = GradleRunner.create().withPluginClasspath().getPluginClasspath()
+            .collect { it.path.replace('\\', '\\\\') } // escape backslashes in Windows paths
```

Comment:
@bmuschko that failed as well with the below error complaining about version. The examples you cited don't use a version so I suspect things should be wired in such a way as to pick the very thing that was just built?

* What went wrong:
Plugin [id: 'com.bmuschko.gradle.docker.DockerRemoteApiPlugin'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Gradle TestKit (classpath: /home/cdancy/github-projects/gradle-docker-plugin/build/classes/main:/home/cdancy/github-projects/gradle-docker-plugin/build/resources/main)
- Gradle Central Plugin Repository (plugin dependency must include a version number for this source)

Relevant code snippet:

    def setup() {
        projectDir = temporaryFolder.root
        buildFile = temporaryFolder.newFile('build.gradle')

        buildFile << """
            plugins {
                id 'com.bmuschko.gradle.docker.DockerRemoteApiPlugin'
            }

			apply plugin: 'java'
			
            repositories {
                mavenCentral()
            }
        """

        setupDockerServerUrl()
        setupDockerCertPath()
        setupDockerPrivateRegistryUrl()
    }